src/main/java/gov/nasa/pds/tools/util/MD5Checksum.java

```diff
@@ -54,23 +62,49 @@ public static String getMD5Checksum(URL url) throws Exception {
    * @throws Exception If an error occurred while calculating the checksum.
    */
   private static byte[] createChecksum(URL url) throws Exception {
-    BufferedInputStream input = null;
-    try {
-      InputStream stream =  Utility.openConnection(url.openConnection());
-      input = new BufferedInputStream(stream);
-      byte[] buffer = new byte[1024];
-      MessageDigest md5 = MessageDigest.getInstance("MD5");
-      int bytesRead = 0;
-      do {
-        bytesRead = input.read(buffer);
-        if (bytesRead > 0) {
-          md5.update(buffer, 0, bytesRead);
+        MessageDigest md5 = MessageDigest.getInstance("MD5");
+        int bytesRead = 0;
+
+        // Use RandomAccessFile to get filesize larger than 2gb
+        File aFile = new File(url.toURI());
+        RandomAccessFile raf = new RandomAccessFile(aFile, "r");
+        raf.seek(0); // Move the pointer to the beginning.
+
+        FileChannel inChannel = raf.getChannel();
+        int bufferSize = 10*1024;   // The buffer size does affect the performance when the file is large.  This has been found to be optimal.
+
+        if (bufferSize > inChannel.size()) {
+            bufferSize = (int) inChannel.size();
         }
-      } while (bytesRead != -1);
-      return md5.digest();
-    } finally {
-      input.close();
-    }
+        ByteBuffer byteBuffer = ByteBuffer.allocate(bufferSize);
```

Comment:
@qchaupds similar to my comment in pds4-jparser, do we think it may make sense to group a lot of these data content checks together into one "walk" through the its of the file? this seems like a 3rd time we are now stepping through the product to check something, which is a 3x performance degradation for very large files. it may not be possible, but worth a discussion.